docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,264 @@
+| chat.rag.context_refinement.prompt | Prompt template for query refinement | See examples below | `--rag-context-prompt` |
+| chat.rag.retriever.top_k | The maximum number of documents to retrieve | `10` | `--retriever-top-k` |
+| chat.rag.retriever.min_score_threshold | The minimum score threshold for chunks to be considered | `0.5` | `--retriever-min-score-threshold` |
+| chat.rag.retriever.filters | Comma separated key-value pairs filters to restrict the search space | See **Notes** | `--retriever-filters` |
+| chat.rag.prompt | Prompt template for RAG-based queries | See examples below | `--rag-prompt` |
+
+Equivalent YAML document for the newly proposed options:
+```yaml
+chat:
+  rag:
+    enabled: false
+    context_refinement:
+      enabled: true
+      prompt: |
+        You are an assistant helping refine user queries for retrieval from a vector database. 
+        The goal is to extract the most relevant information for answering the user's request. 
+        Rewrite this query to focus on the key concepts and details that will retrieve the most contextually 
+        relevant results from the database. 
+        Make it concise and specific, avoiding conversational context or ambiguity.
+        Return the refined query in a single sentence. Avoid any undesired comment or consideration.
+        User query:
+        {{user_query}}
+        Refined query:
+    retriever:
+      top_k: 10
+      min_score_threshold: 0.5
+      filters: 'url=doc.pdf'
+    prompt: |
+      Given the following information, answer the question.
+      Context:
+      {{context}}
+      Question: {{question}}
+      Answer:
+```
+
+**Notes**:
+* Example of `filters` for a `milvuslite` DB:
+```
+--retriever-filters "url=https://en.wikipedia.org/wiki/Mausoleum_at_Halicarnassus"
+```
+which translates to the following dictionary in the retrieval request (*):
+```json
+{
+   "operator":"==",
+   "field":"url",
+   "value":"https://en.wikipedia.org/wiki/Mausoleum_at_Halicarnassus"
+}
+```
+(*) This is for `milvuslite`, other DB can have a different implementation
+
+### 3.6 References
+* [Haystack-DocumentSplitter](https://github.com/deepset-ai/haystack/blob/f0c3692cf2a86c69de8738d53af925500e8a5126/haystack/components/preprocessors/document_splitter.py#L55)
+* [MilvusEmbeddingRetriever](https://github.com/milvus-io/milvus-haystack/blob/77b27de00c2f0278e28b434f4883853a959f5466/src/milvus_haystack/milvus_embedding_retriever.py#L18)
+
+
+### 3.7 Workflow Visualization
+<!-- https://excalidraw.com/#json=hyDopdXaChbf6TNuLGYE0,WkLQBF84vypx0JwAL5iwzA -->
+Ingestion pipeline:
+![ingestion-mvp](../images/ingestion-mvp.png)
+Chat pipeline enriched by RAG context:
+![rag-chat](../images/rag-chat.png)
+
+### 3.8 Proposed Implementation Stack
+The following technologies form the foundation of the proposed solution:
+
+* [Haystack](https://haystack.deepset.ai/): Framework for implementing RAG pipelines and applications.
```

Comment:
We need a dev-doc for this decision, I think.

Reply:
+1

Reply:
To be clear, I am not currently convinced that Haystack is the right choice for us. I did not see a thorough evaluation of the other open source frameworks and given that they are a private company, I have concerns. I am very comfortable with Milvus as it is a part of the LFAI foundation.

Reply:
@franciscojavierarceo you can also comment the related ADR #164